src/main/java/org/isf/medicalstockward/manager/MovWardBrowserManager.java

```diff
@@ -295,4 +296,55 @@ public int compare(MovementForPrint o1, MovementForPrint o2) {
 		}
 	}
 
+	/* * 
+	 * Delete the MovementWard for the specified criteria.
+	 *
+	 * @param movWardToDelete - the movement ward
+	 * @throws OHServiceException if an error occurs retrieving the medical.
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void deleteLastMovementWard(MovementWard movWardToDelete) throws OHServiceException {
+		MovementWard lastMovementWard = ioOperations.getLastMovementWard(movWardToDelete.getWard());
+		if (lastMovementWard.getCode() != movWardToDelete.getCode()) {
+			throw new OHDataValidationException(
+							new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.onlythelastmovementcanbedeleted.msg")));
+		}
+		if (movWardToDelete.getWardTo() != null) {
+			List<MovementWard> lastMovWardWithMed = ioOperations.getMovWardWithWardMedLotDate(movWardToDelete.getWardTo().getCode(), movWardToDelete.getMedical().getCode(), movWardToDelete.getLot().getCode(), movWardToDelete.getDate());
+			if (lastMovWardWithMed.size() == 1) {
```

Comment:
```suggestion
			if (lastMovWardWithMed.size() > 0) {
```

Reply:
no the if the size is greather than 1 then we denied to delete that is the aims

Reply:
sorry my bad, the check should be `== 0` or `.isEmpty()` because the logic is that if we find at least one movement for the same ward, same medical, same lot and after or the same date of the movement we want to delete, then the deletion should be denied.

please note that tests are too fast to really differentiate instants between movements so the `>= :dateFrom"` could result in false positives

Reply:
ok, maybe it is `1` because the query cannot differetiate if it is an incoming or a outgoing movement in the destination ward, so the query will return at least the movement generated from the one we want to delete, right?

Reply:
okay great